### PR TITLE
implement keyword extraction api

### LINKED ISSUE #26: Implement the keyword API
check [API Draft](https://docs.google.com/document/d/1Z19GKOGJVhKviADtvBEPEok53qLszR71e5_4Dz2MTGw/edit#)

this endpoint will again be called from the extension and should handle keyword extraction and return it. Similar to other endpoints should handle edge cases such as bad format, no text, timing out, etc

### PR DESCRIPTION
**Pull request by Michaeltaoma**
resolved #26

Implement keyword extraction api.

Snipaste_2022-11-23_10-03-50

Snipaste_2022-11-22_22-31-01

Keyword extraction might be affected by some HTML tags as they have not been cleaned; we can make text preprocessing into another issue.

### CODE REVIEW

File: frontend/src/components/ArticleSection/index.tsx [hunk 1]
```diff
@@ -37,9 +37,15 @@ export default function Article(props: ArticleProps) {
   };
 
   const generateKeywords = () => {
-    const temp = props.body[0].split(' ');
-    setkeywords(['Examples:'].concat(temp.slice(0, 3)));
-    //TODO: connect with BE
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ text: props.body.join(), numOfKeywords: 3})
```
Comment (Hgjnnf) [inline frontend/src/components/ArticleSection/index.tsx:43]: Is 3 an arbitrary number? Or have we decided on 3 being the best number of keywords to use?

**Review by Hgjnnf — APPROVED** [review]
Looks good to me!

**Review by armintalaie — APPROVED** [review]
for the buggy cases like the tags, if the keywords can't be generated, is that case handled in FE? also could you remove the `keywords:` element? it was just a placeholder. 

great job @Michaeltaoma

File: frontend/src/components/ArticleSection/index.tsx (hunk 1 above)
Comment (Michaeltaoma) [reply frontend/src/components/ArticleSection/index.tsx:43]: 3 is an argument that we can change in the future. Agree with Armin that more keywords might affect the visual appeal.

**Pull request merged**